platform/engineering/design-docs/2020-09-24-package-forms-library.md

```diff
@@ -0,0 +1,217 @@
+
+### Work Estimates
+
+#### Prep work
+
+This involves removing unused forms code and rearranging things so that the forms library doesn't import from app code.
+
+**Estimate:** < 1 day
+
+#### Creating the package
+
+This captures the bulk of the work. Consolidating the forms library code into a single location, fixing broken imports, updating babel and webpack configuration.
+
+**Estimate**: < 4 days
+
+#### Creating the boundaries
+
+Once the package is created, there will still be pieces of code that will need to be shifted around in order to firmly solidfy the boundaries around the forms library.
+
+**Estimate:** < 1 week
+
+#### Updating app code
+
+By this point, the package will already exist, but due to some babel and webpack config it will be masquerading as if it were still spread across `platform/forms` and `platform/forms-system`. This is where each app team updates their code to import directly from the new package.
+
+Once app teams have done this, we will be able to remove those temporary configs and add an ESLint rule to prevent direct imports from `platform/packages/forms-library` in order to make sure that the forms library is treated like a package moving forward.
+
+Actual active time here will be fairly low (< 1 day), and will involve waiting for app teams before we will be able to finish the platform changes.
+
+### Alternatives
+
+See the section [Separate repo](#separate-repo) in Future Work. We are avoiding turning the forms library into its own node module with a separate repo because of the additional overhead involved in publishing, setting up CI pipelines, and general additional repo management. This _may_ be a path for the future.
+Another reason why we don't want to do this right now is that we would lost the advantage of the ESLint configuration that is part of `vets-website`.
+Once this configuration is published we will be able to import it into any repo, and at that point it may make sense to move the forms library out of `vets-website`.
+
+Another alternative would be to move the forms library build out of the overall `vets-website` build. One advantage of this would be saving time, since the forms library could be built once and then all apps could use the cached version. However, on my machine a webpack build takes about 30 seconds, so shaving a few seconds off of this time will not be a huge benefit compared to how long it takes to do a full build with content. The downside of this would be added complexity for managing which forms library build to use (i.e. production, localhost, staging), and CI builds would either have to increase pipeline complexity by dealing with cached builds or lose the time savings altogether. This increase in complexity for marginal time savings does not seem worth it.
```

Comment:
> ...since the forms library could be built once and then all apps could use the cached version.

In your current proposal the forms library package not building itself? If not, this implicitly ties the forms library to the babel and webpack configurations used to build the applications. Does that go against the stated "clear boundary" goal of this forms library package?

> The goal of this project is to define clear boundaries around the collection of code, markup, and styling that is referred to as "the forms library".